sdk/keyvault/Azure.Security.KeyVault.Secrets/MigrationGuide.md

```diff
@@ -0,0 +1,251 @@
+
+The new Key Vault secrets library `Azure.Security.KeyVault.Secrets` provides the ability to share in some of the cross-service improvements made to the Azure development experience, such as using the new `Azure.Identity` library to share a single authentication between clients and a unified diagnostics pipeline offering a common view of the activities across each of the client libraries.
+
+While we believe that there is significant benefit to adopting the new Key Vault secrets library `Azure.Security.KeyVault.Secrets`, it is important to be aware that the previous version `Microsoft.Azure.KeyVault` has not been officially deprecated. It will continue to be supported with security and critical bug fixes. However, it is not under active development and will eventually be deprecated.
+
+## General changes
+
+### Package and namespaces
+
+Package names and the namespace root for the modern Azure client libraries for .NET have changed. Each will follow the pattern `Azure.[Area].[Services]` where the legacy clients followed the pattern `Microsoft.Azure.[Service]`. This provides a quick and accessible means to help understand, at a glance, whether you are using the modern or legacy clients.
+
+In the case of Key Vault, the modern client libraries have packages and namespaces that begin with `Azure.Security.KeyVault` and were released beginning with version 4. The legacy client libraries have packages and namespaces that begin with `Microsoft.Azure.KeyVault` and a version of 3.x.x or below.
+
+### Separate clients
+
+In the interest of simplifying the API we've split `KeyVaultClient` into separate packages and clients:
+
+- `Azure.Security.KeyVault.Certificates` contains `CertificateClient` for certificate management operations.
+- `Azure.Security.KeyVault.Keys` contains `KeyClient` for key management operations and `CryptographyClient` for cryptographic operations.
+- `Azure.Security.KeyVault.Secrets` contains `SecretClient` for secret management operations.
+
+Because [Role-Based Access Control (RBAC)](https://docs.microsoft.com/azure/role-based-access-control/overview) is recommended, we did not implement APIs for Key Vault-managed storage accounts. See [our sample](https://docs.microsoft.com/samples/azure/azure-sdk-for-net/share-link/) for source you can copy into your project if you require managing [Shared Access Signature (SAS)](https://docs.microsoft.com/azure/storage/common/storage-sas-overview) tokens in your application and cannot use RBAC.
```

Comment:
I would move this towards the bottom of the document and keep this part focused on the 3 main clients.

Reply:
What about moving this to a separate heading (h3) at the bottom - after "Deleting secrets" - about SAS tokens and probably just paste this paragraph as-is (or maybe some tweaks to better stand alone). @schaabs, what do you think?

Reply:
Yes I would move it under it's own header at the bottom, but possibly reword the first sentence a bit.

> Because  [Role-Based Access Control (RBAC)](https://docs.microsoft.com/azure/role-based-access-control/overview) is now recommended for managing access to Azure Storage accounts, the APIs for Key Vault managed storage accounts are no longer available in version 4 of the Key Vault client library.

Thoughts?

Reply:
Sure, though I still want to link the sample but I can tweak it all a bit to better stand alone. Thanks, all!